packages/next/src/client/components/react-dev-overlay/server/middleware-turbopack.ts

```diff
@@ -171,7 +171,11 @@ export function getSourceMapMiddleware(project: Project, distDir: string) {
 
     try {
       if (filename.startsWith('/_next/static')) {
-        filename = path.join(distDir, filename.replace(/^\/_next\//, ''))
+        filename = path.join(
+          distDir,
+          // /_next/static/chunks/%5Bproject%5D... => static/chunks/[project]...
+          decodeURIComponent(filename.replace(/^\/_next\//, ''))
```

Comment:
Might be safer to do this earlier when we read it from search params in case we add branches later. 

I'm actually confused why we need to do this anyway. Shouldn't `searchParams.get()` take care of that? Or is the `filename` somehow encoded twice?

Reply:
Exactly, it's double-encoded. I think we should follow up on that with the Turbopack team, as I don't think this mix of encoded and not encoded links is right:

```
<link rel="preload" as="script" fetchPriority="low" href="/_next/static/chunks/%5Bproject%5D_packages_next_dist_compiled_react-dom-experimental_487273._.js"/>
<script src="/_next/static/chunks/%5Bproject%5D_packages_next_dist_compiled_5b81ff._.js" async=""></script>
<script src="/_next/static/chunks/%5Bproject%5D_packages_next_dist_client_ca77f6._.js" async=""></script>
<script src="/_next/static/chunks/%5Bproject%5D_packages_next_dist_082c14._.js" async=""></script>
<script src="/_next/static/chunks/de9e7_%40swc_helpers_cjs_6ab2e1._.js" async=""></script>
<script src="/_next/static/chunks/%5Bturbopack%5D_browser_dev_hmr-client_hmr-client_ts_5dfc9a._.js" async=""></script>
<script src="/_next/static/chunks/%5Bproject%5D_test_development_app-dir_source-mapping_c4c472._.js" async=""></script>
<script src="/_next/static/chunks/%5Bproject%5D_test_development_app-dir_source-mapping_bc5604._.js" async=""></script>
<script src="/_next/static/chunks/[project]__cfa445._.js" async=""></script>
<script src="/_next/static/chunks/app_layout_tsx_6092de._.js" async=""></script>
<script src="/_next/static/chunks/[project]__48591b._.js" async=""></script>
<script src="/_next/static/chunks/app_client_page_tsx_06de4e._.js" async=""></script>
<script src="/_next/static/chunks/%5Bproject%5D__cfa445._.js" async=""></script>
<script src="/_next/static/chunks/%5Bproject%5D__48591b._.js" async=""></script>
<script src="/_next/static/chunks/[project]_packages_next_dist_build_polyfills_polyfill-nomodule.js" noModule=""></script>
```

I just noticed that this even leads to duplicates being loaded.